src/components/AdminPage/ConfirmDialog.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Confirm } from 'semantic-ui-react';
+
+class ConfirmDialog extends Component {
+  constructor(props) {
+    super(props);
+    this.toggleDialog = this.toggleDialog.bind(this);
+    this.deleteArticle = this.deleteArticle.bind(this);
+    this.state = { isOpen: false };
+  }
+
+  deleteArticle() {
+    const { id } = this.props;
+    this.deleteArticle(id);
+    this.toggleDialog();
+  }
+
+  toggleDialog() {
+    const { isOpen } = this.state;
+    this.setState({ isOpen: !isOpen });
+  }
+
+  render() {
+    const { isOpen } = this.state;
+    return (
+      <div>
+        <Button
+          onClick={this.toggleDialog}
+          color="red"
+          size="mini"
+          icon="delete"
+        />
+        <Confirm
+          open={isOpen}
+          onCancel={this.toggleDialog}
+          onConfirm={this.deleteArticle}
```

Comment:
The library's implementation is the same as what I recommended above. They take a function to call `onConfirm` and one `onCancel`. This tbh makes your `ConfirmDialog` component a bit redundant since we can just use their one.

One thing you can do which I think is a good design is to keep is as a wrapper. Essentially make it so its used like this:
```
<ConfirmDialog
    onConfirm={yourFunction}
    onCancel={yourOtherFunction}
    trigger={<Button />}
/>
```
The trigger will be what is clicked to toggle the dialog, which is better than hardcoding the button details.